tree/tree/src/TBasket.cxx

```diff
@@ -769,24 +771,21 @@ void TBasket::SetWriteMode()
 void TBasket::Streamer(TBuffer &b)
 {
    char flag;
+   bool hasIOBits = false;
    if (b.IsReading()) {
       TKey::Streamer(b); //this must be first
       Version_t v = b.ReadVersion();
       b >> fBufferSize;
       b >> fNevBufSize;
       if (fNevBufSize < 0) {
-         Error("Streamer","The value of fNevBufSize is incorrect (%d) ; trying to recover by setting it to zero",fNevBufSize);
```

Comment:
Yes, it is replaced by https://github.com/root-project/root/pull/959/files#diff-244ab93230dcfa0c91eb218428db7be6R815

Basically, if you have a negative value here (corruption of the upper bit), then we apply  checks to the serialized `fIOBits` -- the upper bit there also must be zero.

So, before, we can detect a single bit flip; afterward, we can detect a single bit flip.

Reply:
>  the upper bit there also must be zero.

Humm ... this is not documented, is it?  I.e. we 'might' have 255 variations (we won't but nothing yet prevent the use of the upper bit :)

> So, before, we can detect a single bit flip; afterward, we can detect a single bit flip.

Fair enough, let's make it clear that the location of the test switched in comments or commits.

Reply:
One of the primary side effect was to avoid executing:
```
tree/tree/src/TBasket.cxx:   if (fNevBufSize) {
tree/tree/src/TBasket.cxx:      fEntryOffset = new Int_t[fNevBufSize];
```
with a crazy fNevBufSize value.  I did not follow the new code enough (with your new test on IsZombie) to verify whether the execution of that line is avoided in case of seeming corruption.

> So, before, we can detect a single bit flip; afterward, we can detect a single bit flip.

Actually not quite.  With a file that **is** using the extensions, a crazy value of fNevBufSize will no longer be detected and can still lead to a large memory allocation (half as large as before but still ....)